unit_tests/src/test_rcMisc_functions.cpp

```diff
@@ -0,0 +1,289 @@
+#include "catch.hpp"
+
+#include "rcMisc.h"
+
+#include "fmt/format.h"
+#include "fmt/chrono.h"
+
+#include <cstring>
+#include <map>
+#include <string>
+#include <string_view>
+
+struct input
+{
+    std::string_view current_time;
+    std::string_view frequency;
+};
+
+struct output
+{
+    int expected_action_code;
+    std::string_view expected_next_time;
+    std::string_view expected_frequency;
+};
+
+auto test_execution_frequency_handler(const input& _input, const output& _output)
+{
+    char current_time[TIME_LEN]{};
+    std::strcpy(current_time, _input.current_time.data());
+
+    char frequency[128]{};
+    std::strcpy(frequency, _input.frequency.data());
+
+    char next_time[TIME_LEN]{};
+
+    CHECK(getNextRepeatTime(current_time, frequency, next_time) == _output.expected_action_code);
+    CHECK(next_time == _output.expected_next_time);
+    CHECK(frequency == _output.expected_frequency);
+}
+
+template <typename Timepoint>
+auto make_repeat_until_time_string(int _nnnn,
+                                   const std::string_view _U,
+                                   const Timepoint _tp) -> std::string
+{
+    return fmt::format("{}{} REPEAT UNTIL {}", _nnnn, _U, std::chrono::system_clock::to_time_t(_tp));
+}
+
+template <typename Timepoint>
+auto make_repeat_until_success_or_until_time_string(int _nnnn,
+                                                    const std::string_view _U,
+                                                    const Timepoint _tp) -> std::string
+{
+    return fmt::format("{}{} REPEAT UNTIL SUCCESS OR UNTIL {}", _nnnn, _U, std::chrono::system_clock::to_time_t(_tp));
+}
+
+#if 0
```

Comment:
maybe mention #5964 in a comment here or at each `if 0`

Reply:
will do

Reply:
done